Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/NonprofitViewController.m

```diff
@@ -7,17 +7,42 @@
 //
 
 #import "NonprofitViewController.h"
+#import "Nonprofit.h"
+#import "Utils.h"
+#import "User.h"
+
 
 @interface NonprofitViewController ()
 @property (weak, nonatomic) IBOutlet UIButton *favoriteButton;
+@property (weak, nonatomic) IBOutlet UIImageView *nonprofitHeaderPic;
+@property (weak, nonatomic) IBOutlet UIImageView *nonprofitProfilePic;
+@property (weak, nonatomic) IBOutlet UILabel *nonprofitNameLabel;
+@property (weak, nonatomic) IBOutlet UILabel *nonprofitCategoryLabel;
+@property (weak, nonatomic) IBOutlet UILabel *nonprofitWebsiteLabel;
+@property (weak, nonatomic) IBOutlet UIImageView *verifiedImageView;
+@property (weak, nonatomic) IBOutlet UILabel *totalValueDonatedLabel;
+@property (weak, nonatomic) IBOutlet UILabel *nonprofitDescriptionLabel;
 
 @end
 
 @implementation NonprofitViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    [self setupView];
+}
+
+- (void)setupView {
+    self.nonprofitHeaderPic.image = [Utils getImageFromPFFile:self.nonprofit.headerPicFile];
+    self.nonprofitProfilePic.image = [Utils getImageFromPFFile:self.nonprofit.profilePicFile];
+    self.favoriteButton.selected = [[[User currentUser] favoriteNonprofits] containsObject:self.nonprofit];
+    self.totalValueDonatedLabel.text = [NSString stringWithFormat:@"$%0.2f", self.nonprofit.totalDonationsValue];
```

Comment:
The favorite and total value logic is very simple, but could also be encapsulated in the `Nonprofit` and `User` models instead of duplicated. You could define methods like `- (BOOL)isFavoriteNonprofit:(Nonprofit *)nonprofit;` in the `User` model, and `- (NSString *)formattedTotalDonations;` in the `Nonprofit` class and reuse everywhere.